### PR TITLE
chore: Docker SDK minimum version should be 3.7.0

### PR DESCRIPTION
**Pull request by sanathkr**
To support Docker Proxy Config (https://github.com/awslabs/aws-sam-cli/pull/1196)

*Issue #, if available:*

*Description of changes:*

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sanathkr** [comment]
I reviewed the changelog of Docker-Py library (https://github.com/docker/docker-py/releases) since v3.3.0. No breaking changes and no warnings or anything. The list of features & bug fixes look benign to me.

File: requirements/base.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ boto3~=1.9, >=1.9.56
 PyYAML~=3.12
 cookiecutter~=1.6.0
 aws-sam-translator==1.10.0
-docker>=3.3.0
+docker>=3.7.0
```
Comment (jfuss) [inline requirements/base.txt]: If we are not planning to update to the latest, can we bump this (and pin) to the latest 3.x [3.7.2](https://github.com/docker/docker-py/releases/tag/3.7.2)

Comment (sanathkr) [reply requirements/base.txt]: Actually, we should stick to ~3.7.0 which will upgrade patch versions...

Comment (jfuss) [reply requirements/base.txt]: That means build are not deterministic though.

Comment (sanathkr) [reply requirements/base.txt]: Yes, but patch versions shouldn't break..

**Pull request merged**